client.js

```diff
@@ -326,34 +229,84 @@ async function downloadMod() {
 	}
 }
 
+/**
```

Comment:
Question, maybe this should go into the lib folder as utility methods? So we don't have big fat client.js with all the code and something more manageable?